src/formatters/Utils.cc

```diff
@@ -48,6 +50,54 @@ using lsst::daf::persistence::LogicalLocation;
 namespace lsst {
 namespace afw {
 namespace formatters {
+namespace {
+
+/**
+Format a PropertySet into a FITS header string (exactly 80 characters per "card", no line terminator)
+
+@param[in] paramNames  Names of properties to format
+@param[in] prop  Properties to format
+
+@todo Once we stop using the old WCS class we can remove the PropertySet version of formatFitsProperties
+and simplify this function to not worry about dotted names.
+*/
+std::string formatFitsPropertiesImpl(std::vector<std::string> const& paramNames,
+                                     daf::base::PropertySet const& prop) {
+    std::ostringstream result;
+    for (auto const & fullName : paramNames) {
+        std::size_t lastPeriod = fullName.rfind(char('.'));
+        auto name = (lastPeriod == std::string::npos) ? fullName : fullName.substr(lastPeriod + 1);
+        std::type_info const& type = prop.typeOf(name);
+
+        std::string out = "";
+        if (name.size() > 8) {  // Oh dear; too long for a FITS keyword
+            out += "HIERARCH = " + name;
+        } else {
+            out = (boost::format("%-8s= ") % name).str();
+        }
+
+        if (type == typeid(int)) {
+            out += (boost::format("%20d") % prop.get<int>(name)).str();
+        } else if (type == typeid(double)) {
+            out += (boost::format("%20.15g") % prop.get<double>(name)).str();
+        } else if (type == typeid(std::string)) {
+            out += (boost::format("'%-67s' ") % prop.get<std::string>(name)).str();
```

Comment:
This is pre-existing code (adapted to an order-preserving variant). For its purposes we don't care about CONTINUE cards or COMMENT cards or any of that. We're trying to get WCS numbers out.